loggibud/v1/baselines/task1/lkh_3.py

```diff
@@ -45,85 +50,48 @@ def solve(
 
     params = params or LKHParams()
 
-    logger.info("Converting instance into a TSPLIB file")
-    convert_instance_file(instance, params)
+    conversion_params = TSPLIBConversionParams(osrm_config=params.osrm_config)
+    tsplib_instance = to_tsplib(instance, conversion_params)
 
-    logger.info("Calling LKH external solver")
-    solve_lkh(instance, params)
+    # LKH solution params, for details check the LKH documentation.
+    lkh_params = dict(
+        mtsp_objective="MINSUM",
+        runs=params.num_runs,
+        time_limit=params.time_limit_s,
+        vehicles=_get_num_vehicles(instance),
+    )
 
-    logger.info("Reading result from output files")
-    solution = read_solution(instance, params)
+    current_path = os.path.dirname(os.path.abspath(__file__))
+    lkh_solution = lkh.solve(
+        f"{current_path}/LKH", tsplib_instance, **lkh_params
+    )
 
-    logger.info("Cleaning up temporary files")
-    remove_auxiliary_files(instance)
+    solution = _unwrap_lkh_solution(instance, lkh_solution)
 
     return solution
 
 
-def convert_instance_file(instance: CVRPInstance, params: LKHParams) -> None:
-    """
-    The LKH-3 solver requires two files:
-        - A .par file with parameters for the solver;
-        - A .vrp file with information about the instance.
-    The vrp file is created with the original TSPLIB format, and the .par one,
-    which is specific for the LKH-3 solver, is created here.
-    These files will have the instance name to prevent conflicts in case the
-    solver runs in parallel solving multiple instances.
-    """
-
-    auxiliary_file_names = _get_auxiliary_file_names(instance)
-    to_tsplib(instance, file_name=auxiliary_file_names["input_vrp_file"])
-
-    # For the Asymetric CVRP, it only respects the capacity if the number of
-    # vehicles is explicitly provided
-    num_vehicles = _get_num_vehicles(instance)
-    with open(auxiliary_file_names["input_par_file"], "w") as f:
-        f.write(
-            "SPECIAL\n"
-            f"PROBLEM_FILE = {auxiliary_file_names['input_vrp_file']}\n"
-            "MTSP_OBJECTIVE = MINSUM\n"
-            f"RUNS = {params.num_runs}\n"
-            f"TOUR_FILE = {auxiliary_file_names['output_tour_file']}\n"
-            f"TIME_LIMIT = {params.time_limit_s}\n"
-            f"VEHICLES = {num_vehicles}"
-        )
-
-
-def solve_lkh(instance: CVRPInstance, params: LKHParams) -> None:
-    """Call the C solver and generate output files"""
-
-    auxiliary_file_names = _get_auxiliary_file_names(instance)
-    arguments = (
-        "./loggibud/v1/baselines/task1/LKH",
-        auxiliary_file_names["input_par_file"],
-    )
-    popen = subprocess.Popen(arguments, stdout=subprocess.PIPE)
-    popen.wait()  # run the solver in the background
-
-
-def read_solution(instance: CVRPInstance, params: LKHParams) -> CVRPSolution:
+def _unwrap_lkh_solution(
+    instance: CVRPInstance, lkh_solution: list
```

Comment:
Nitpick: Is `lkh_solution` a list of indices? If so, I suggest using `List` from `typing`:

```suggestion
    instance: CVRPInstance, lkh_solution: List[int]
```

(of course, add `from typing import List` at the preamble)